File: InsertionSort.java


public class InsertionSort {

    // Method to perform Insertion Sort
    public static void insertionSort(int[] array) {

        // Start from the second element
        for (int i = 1; i < array.length; i++) {

            // Store the current element
            int key = array[i];

            // Previous index
            int j = i - 1;

            // Shift elements that are greater than key
            // to one position ahead
            while (j >= 0 && array[j] > key) {

                array[j + 1] = array[j];
                j--;

            }

            // Insert the key into its correct position
            array[j + 1] = key;

        }

    }

    // Method to print the array
    public static void printArray(int[] array) {

        for (int number : array) {
            System.out.print(number + " ");
        }

        System.out.println();
    }

    public static void main(String[] args) {

        // Unsorted array
        int[] numbers = {12, 11, 13, 5, 6};

        System.out.println("Before Sorting:");
        printArray(numbers);

        // Call Insertion Sort
        insertionSort(numbers);

        System.out.println("After Sorting:");
        printArray(numbers);

    }
}

File: InsertionSort.c
#include <stdio.h>

int main() {
    int arr[100];
    int n, i, j, key;

    // Input array size
    printf("Enter the number of elements: ");
    scanf("%d", &n);

    // Input array elements
    printf("Enter %d elements:\n", n);
    for(i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    // Insertion Sort
    for(i = 1; i < n; i++) {

        key = arr[i];
        j = i - 1;

        // Move elements greater than key one position ahead
        while(j >= 0 && arr[j] > key) {
            arr[j + 1] = arr[j];
            j--;
        }

        arr[j + 1] = key;
    }

    // Display sorted array
    printf("\nSorted Array (Ascending Order):\n");

    for(i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }

    printf("\n");

    return 0;
}
